service/backend/common/construction.go

```diff
@@ -0,0 +1,347 @@
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/cb58"
+	"github.com/ava-labs/avalanchego/utils/crypto"
+	"github.com/ava-labs/avalanchego/utils/formatting/address"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/coinbase/rosetta-sdk-go/parser"
+	"github.com/coinbase/rosetta-sdk-go/types"
+
+	"github.com/ava-labs/avalanche-rosetta/mapper"
+	pmapper "github.com/ava-labs/avalanche-rosetta/mapper/pchain"
+	"github.com/ava-labs/avalanche-rosetta/service"
+)
+
+var (
+	errNoOperationsToMatch      = errors.New("no operations were passed to match")
+	errInvalidInput             = errors.New("invalid input")
+	errInvalidInputSignatureLen = errors.New("input signature length doesn't match credentials needed")
+	errInsufficientSignatures   = errors.New("insufficient signatures")
+	errInvalidSignatureLen      = errors.New("invalid signature length")
+)
+
+func DeriveBech32Address(fac *crypto.FactorySECP256K1R, chainIDAlias string, req *types.ConstructionDeriveRequest) (*types.ConstructionDeriveResponse, *types.Error) {
```

Comment:
We should add comments to anything that's exported (what the function does, invariants, etc)

Reply:
Sure, I can add some comments.